libs/payments/ui/src/lib/nestapp/nextjs-actions.service.ts

```diff
@@ -114,9 +114,11 @@ export class NextJSActionsService {
   async checkoutCartWithPaypal(args: CheckoutCartWithPaypalActionArgs) {
     await new Validator().validateOrReject(args);
 
-    await this.cartService.checkoutCartWithPaypal(
+    await this.cartService.setCartProcessing(args.cartId);
```

Comment:
What about updating the cart to `processing` here as part of the nextjs-action service?

Reply:
I'd like the checkoutCartWithPaypal method to be responsible for these state changes. checkoutCartWithPaypal should throw an error if called on a cart in the processing state.